lib/overcommit/installer.rb

```diff
@@ -118,6 +125,32 @@ def can_replace_file?(file)
         overcommit_hook?(file)
     end
 
+    def preserve_old_hooks
+      return unless File.directory?(hooks_path)
+
+      ensure_directory(old_hooks_path)
+      Overcommit::Utils.supported_hook_types.each do |hook_type|
```

Comment:
Ah, I forgot. Another issue with just moving everything to `.git/old-hooks` on install and then doing `rm -rf .git/hooks && mv .git/old-hooks .git/hooks` on uninstall is that the user may have thrown some custom hooks into `.git/hooks` in the meantime, and we don't want to clobber those.